wait/host.go

```diff
@@ -50,10 +49,10 @@ func (a *HostAwaitility) GetMasterUserRecord(name string) *toolchainv1alpha1.Mas
 	return mur
 }
 
-// WaitForMasterUserRecord waits until there is MasterUserRecord available with the given name and meeting the set of given wait-conditions
-func (a *HostAwaitility) WaitForMurConditions(name string, waitCond ...MurWaitCondition) error {
-	return wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
-		mur := &toolchainv1alpha1.MasterUserRecord{}
+// WaitForMurConditions waits until there is MasterUserRecord available with the given name and meeting the set of given wait-conditions
+func (a *HostAwaitility) WaitForMurConditions(name string, waitCond ...MurWaitCondition) (*toolchainv1alpha1.MasterUserRecord, error) {
```

Comment:
well, to be honest, I believe that it's more idiomatic to return both the resource that we expect/wait for along with an error if it was not found, rather than make 2 subsequent calls.

Reply:
I'm ok with making this change in a separate PR, and apply the same pattern to all the other `WaitFor...` funcs for the sake of consistency, btw.

Reply:
it's fine, this comment was more or less proposal than a requested change ;-). 
agree that all the wait functions could return the object if it is found, so if you want, you can change it for all other wait functions in a separated PR, I'm fine with it ;-)

Reply:
great! Let me do that, then ;)

Reply:
addressed in https://github.com/codeready-toolchain/toolchain-e2e/pull/31